src/dotnet/UILanguageOverride.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Globalization;
+
+namespace Microsoft.DotNet.Cli
+{
+    internal static class UILanguageOverride
+    {
+        private const string DOTNET_CLI_UI_LANGUAGE = nameof(DOTNET_CLI_UI_LANGUAGE);
+        private const string VSLANG = nameof(VSLANG);
+        private const string PreferredUILang = nameof(PreferredUILang);
+
+        public static void Setup()
+        {
+            CultureInfo language = GetOverriddenUILanguage();
+            if (language == null)
+            {
+                return;
+            }
+
+            // Make the current process respect the override.
+            CultureInfo.DefaultThreadCurrentUICulture = language;
```

Comment:
Same here about the comment. I would wrap this in a method called MakeCurrentProcessRespectOverride().